projects/js-packages/connection/state/actions.jsx

```diff
@@ -20,6 +35,34 @@ const connectionStatusActions = {
 	setUserIsConnecting: isConnecting => {
 		return { type: SET_USER_IS_CONNECTING, isConnecting };
 	},
+	setRegistrationError: registrationError => {
+		return { type: SET_REGISTRATION_ERROR, registrationError };
+	},
+	clearRegistrationError: () => {
+		return { type: CLEAR_REGISTRATION_ERROR };
+	},
+	setAuthorizationUrl: authorizationUrl => {
+		return { type: SET_AUTHORIZATION_URL, authorizationUrl };
+	},
+	*registerSite( registrationNonce, redirectUri ) {
+		yield dispatch( STORE_ID ).clearRegistrationError();
+		yield dispatch( STORE_ID ).setSiteIsRegistering( true );
+		yield restApi
+			.registerSite( registrationNonce, redirectUri )
+			.then( response => {
+				dispatch( STORE_ID ).setSiteIsRegistering( false );
+				dispatch( STORE_ID ).setConnectionStatus( { isRegistered: true } );
```

Comment:
> Will this result in connectionStatus only having a single isRegistered property?

No! It will only change the value of `isRegistered` and keep the other as they are. We could have more specific actions... but I'm not sure whether this will simplify or complicate things, because we'd have to create an action for each property...

Reply:
> No! It will only change the value of isRegistered and keep the other as they are. 

Nice!

> but I'm not sure whether this will simplify or complicate things, because we'd have to create an action for each property...

I think I'd prefer this approach as it would allow us to treat the whole `connectionStatus` object as a black box and make it easier to adjust to backend changes imho.  However we can further discuss it and address it in a future PR if needed.